### ISSUE TITLE
Rewrites all history instead of just edited commits

### ISSUE THREAD

**Issue by RaeesBhatti** [PotatoLabs/git-redate#3]
When used to edit only a number of previous commits history e.g. `git redate -c 5`. While it will change date for the ones specified, it will rewrite history (change hashes) for all commits instead of just the last five.

**Comment by EdmundMai** [comment]
Yes this is true, although it technically doesn't do anything to the older commits if you don't change the date. 

Has this affected you negatively at all (besides being slower than if it changed the dates for only the 5 commits)?

**Comment by RaeesBhatti** [comment]
When the project is hosted on sites like GitHub where you releases are linked with commit hashes, the releases don't point to correct history anymore.
Also the contribution count on your profile adds up, e.g. every time you rewrite history and upload, it will think that the contributions are new and add the number up. This makes other days much dimmer.

**Comment by EdmundMai** [comment]
Hm, ok. I'll look into it

**Comment by Sweeper777** [comment]
+1

**Comment by SiteRelEnby** [comment]
Fixed in our fork, if anyone else comes across this issue.

https://github.com/SiteRelEnby/git-redate
